src/server/pps/cmds/cmds.go

```diff
@@ -1579,6 +1579,51 @@ func Cmds(mainCtx context.Context, pachCtx *config.Context, pachctlCfg *pachctl.
 	//inspectDefaults.Flags().StringVar(&project, "project", project, "Inspect project defaults.")
 	commands = append(commands, cmdutil.CreateAliases(inspectDefaults, "inspect defaults"))
 
+	var pathname string
+	createDefaults := &cobra.Command{
+		Use:   "{{alias}} [--cluster]",
+		Short: "Set cluster defaults.",
+		Long:  "Set cluster defaults.",
+		Run: cmdutil.RunFixedArgs(0, func(args []string) error {
+			if cluster {
+				r, err := fileIndicatorToReadCloser(pathname)
+				if err != nil {
+					return errors.Wrapf(err, "could not open path %q for reading", pathname)
+				}
+				b, err := io.ReadAll(r)
+				if err != nil {
+					return errors.Wrapf(err, "could not read from %q", pathname)
+				}
+				b = bytes.TrimSpace(b) // remove leading & trailing whitespace
+				// validate that the provided defaults parse
+				var cd pps.ClusterDefaults
+				if err := protojson.Unmarshal(b, &cd); err != nil {
+					return errors.Wrapf(err, "invalid cluster defaults")
+				}
+				var req pps.SetClusterDefaultsRequest
+				req.ClusterDefaultsJson = string(b)
+
+				client, err := pachctlCfg.NewOnUserMachine(mainCtx, false)
+				if err != nil {
+					return err
+				}
+				defer client.Close()
+
+				if _, err := client.PpsAPIClient.SetClusterDefaults(mainCtx, &pps.SetClusterDefaultsRequest{
+					ClusterDefaultsJson: string(b),
+				}); err != nil {
+					return errors.Wrap(err, "could not set cluster defaults")
+				}
+				return nil
+			}
+			return errors.New("--cluster must be specified")
```

Comment:
Just to clarify, is the intent to later add pipeline defaults? But for now, we only support setting cluster defaults?

Reply:
We’ll add _project_ defaults in a future epic.